skyportal/handlers/api/image_analysis.py

```diff
@@ -0,0 +1,451 @@
+from astropy.time import Time
+import astropy.units as u
+
+import base64
+import numpy as np
+from sqlalchemy.orm import sessionmaker, scoped_session
+import tempfile
+from tqdm.auto import tqdm
+from tornado.ioloop import IOLoop
+import shutil
+
+from .photometry import add_external_photometry
+from ..base import BaseHandler
+from ...models import Instrument, User, DBSession
+
+_, cfg = load_env()
+
+log = make_log('image_analysis')
+
+try:
+    if cfg['image_analysis'] is True:
+        import astroscrappy
+        from stdpipe import (
+            astrometry,
+            photometry,
+            catalogs,
+            cutouts,
+            templates,
+            plots,
+            pipeline,
+            psf,
+            utils,
+        )
+except Exception as e:
+    log(e)
+
+try:
+    if cfg['image_analysis'] is True:
+        # remove any temp dir starting with 'sex'
+        for dir in os.listdir('/tmp'):
+            if dir.startswith('sex') or dir.startswith('psfex'):
+                shutil.rmtree(os.path.join('/tmp', dir))
+except Exception as e:
+    log(e)
+
+
+def spherical_match(ra1, dec1, ra2, dec2, sr=1 / 3600):
+    """Positional match on the sphere for two lists of coordinates.
+
+    Aimed to be a direct replacement for :func:`esutil.htm.HTM.match` method with :code:`maxmatch=0`.
+
+    :param ra1: First set of points RA
+    :param dec1: First set of points Dec
+    :param ra2: Second set of points RA
+    :param dec2: Second set of points Dec
+    :param sr: Maximal acceptable pair distance to be considered a match, in degrees
+    :returns: Two parallel sets of indices corresponding to matches from first and second lists, along with the pairwise distances in degrees
+
+    """
+    try:
+        idx1, idx2, dist, _ = search_around_sky(
+            SkyCoord(ra1, dec1, unit='deg'), SkyCoord(ra2, dec2, unit='deg'), sr * u.deg
+        )
+
+        dist = dist.deg  # convert to degrees
+
+        return idx1, idx2, dist
+    except Exception as e:
+        raise e
+
+
+def spherical_distance(ra1, dec1, ra2, dec2):
+    try:
+        x = np.sin(np.deg2rad((ra1 - ra2) / 2))
+        x *= x
+        y = np.sin(np.deg2rad((dec1 - dec2) / 2))
+        y *= y
+
+        z = np.cos(np.deg2rad((dec1 + dec2) / 2))
+        z *= z
+
+        return np.rad2deg(2 * np.arcsin(np.sqrt(x * (z - y) + y)))
+    except Exception as e:
+        raise e
+
+
+def reduce_image(image, header, obj_id, instrument_id, user_id, detect_cosmics=False):
+
+    try:
+        workdir_sextractor_obj = tempfile.mkdtemp(prefix='sex')
+        workdir_sextractor_obj1 = tempfile.mkdtemp(prefix='sex1')
+        workdir_psfex = tempfile.mkdtemp(prefix='psfex')
+        Session = scoped_session(
+            sessionmaker(bind=DBSession.session_factory.kw["bind"])
+        )
+        session = Session()
+
+        instrument = session.query(Instrument).get(instrument_id)
+        user = session.query(User).get(user_id)
```

Comment:
Also we can do this using `select` instead of `query` like in the other handlers.